int/api/wallet/import.go

```diff
@@ -59,13 +59,34 @@ func (c *wImport) Handle(params operations.MgmtWalletImportParams) middleware.Re
 			})
 	}
 
-	newWallet, err := wallet.Imported(walletName, privateKey)
-	if err != nil {
-		return operations.NewMgmtWalletCreateInternalServerError().WithPayload(
-			&models.Error{
-				Code:    errorCodeWalletCreateNew,
-				Message: err.Error(),
-			})
+	newWallet, RequestError := wallet.Imported(walletName, privateKey)
+	if RequestError.Err != nil {
+		switch {
+		case RequestError.StatusCode == wallet.ImportedAlreadyImported:
+			return operations.NewMgmtWalletCreateInternalServerError().WithPayload(
+				&models.Error{
+					Code:    errorCodeWalletAlreadyImported,
+					Message: RequestError.Err.Error(),
+				})
+		case RequestError.StatusCode == wallet.ImportedEncodingB58Error:
+			return operations.NewMgmtWalletCreateInternalServerError().WithPayload(
+				&models.Error{
+					Code:    errorCodeWalletEncodingB58E,
+					Message: RequestError.Err.Error(),
+				})
+		case RequestError.StatusCode == wallet.ImportedLoadingWalletsError:
+			return operations.NewMgmtWalletCreateInternalServerError().WithPayload(
+				&models.Error{
+					Code:    errorCodeWalletLoadingWallets,
+					Message: RequestError.Err.Error(),
+				})
+		default:
+			return operations.NewMgmtWalletCreateInternalServerError().WithPayload(
+				&models.Error{
+					Code:    errorCodeWalletCreateNew,
+					Message: RequestError.Err.Error(),
+				})
```

Comment:
can you factorize this ?! something like NewWalletError( code, RequestError)

Reply:
cannot use nil as RequestError value, since type RequestError struct {
	StatusCode int
	Err        error
}